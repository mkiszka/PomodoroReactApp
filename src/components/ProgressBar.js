import React from "react";

function ProgressBar({ percent = 0, trackRemaining = false, className }) {

    return (
        <div className={
            `ProgressBar ProgressBar_trackRemaining_${trackRemaining} ${className} `
        }
            style={{ "--width": `${percent}%` }}>

        </div>
    );
}

export default ProgressBar;