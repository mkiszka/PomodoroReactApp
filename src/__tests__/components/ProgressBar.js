import TestRenderer from 'react-test-renderer';
import ProgressBar from '../../components/ProgressBar';

describe('ProgressBar ', () => {
    let progressBar = null;
    let progressBarJSON = null;

    describe('when none of parameters was set, should ', () => {
        beforeAll(() => {
            progressBar = TestRenderer.create(<ProgressBar />);
            progressBarJSON = progressBar.toJSON();
        })
        it('not contain undefined className', () => {
            expect(progressBarJSON.props.className).not.toMatch(/undefined/);
        });
        it('looks always the same',()=> {
            expect(progressBarJSON).toMatchSnapshot();
        });
        /*
        {
              type: 'div',
              props: {
                className: 'ProgressBar ProgressBar_trackRemaining_false undefined ',
                style: { '--width': '0%', '--background-color': 'red' }
              },
              children: null
            }
            */
/*
        it(' .... ', () => {
            console.log(progressBar.toJSON());
            expect(progressBar.toJSON()).toBe();
        })
        */
    });
});