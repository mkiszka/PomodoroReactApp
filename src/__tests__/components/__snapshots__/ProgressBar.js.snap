// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar  when none of parameters was set, should  looks always the same 1`] = `
<div
  className="ProgressBar ProgressBar_trackRemaining_false  "
  data-testid="ProgressBar-div"
  style={
    Object {
      "--background-color": "red",
      "--width": "0%",
    }
  }
/>
`;
